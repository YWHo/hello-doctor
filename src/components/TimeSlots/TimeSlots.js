import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import dayjs from 'dayjs';

const ContainerOuter = styled.div`
  display: block;
  margin-top: 16px;
  padding-top: 12px;
  height: 37.95px;
  width: 100%;
  overflow: hidden;
  white-space: nowrap;
`;

const ContainerInner = styled.div`
  display: block;
  height: 42px;
  overflow-x: auto;
  overflow-y: hidden;
  padding: 0;
  white-space: nowrap;
`;

const ContainerNoAppointment = styled.div`
  margin-top: 25.19px;
  height: 38.37px;
  padding-right: 12px;
  width: 100%;
`;

const NoAppointment = styled.div`
  background-color: #f7f7f7;
  font-size: 16px;
  line-height: 38.37px;
  height: 38.37px;
  width: 100%;
  text-align: center;
  vertical-align: middle;
`;

export function TimeSlots(props) {
  const { availableSlots = {} } = props;
  const slots = Object.keys(availableSlots).map(key =>
    dayjs(availableSlots[key]).format('HH:mm')
  );

  if (slots.length === 0) {
    return (
      <ContainerNoAppointment>
        <NoAppointment>No appointment available for this date</NoAppointment>
      </ContainerNoAppointment>
    );
  } else {
    return (
      <ContainerOuter>
        <ContainerInner>{slots.join(' | ')}</ContainerInner>
      </ContainerOuter>
    );
  }
}

const mapStateToProps = state => {
  return {};
};

function mapDispatchToProps(dispatch) {
  return {};
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TimeSlots);
