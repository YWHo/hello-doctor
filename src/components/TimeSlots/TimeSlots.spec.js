import React from 'react';
import { shallow } from 'enzyme';
import { TimeSlots } from './TimeSlots';

describe('DaySlots', () => {
  it('render "No Appointment Available"', () => {
    const wrapper = shallow(<TimeSlots />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render based on the given schedule', () => {
    const schedules = {
      '546ad6da-1fe7-4081-a820-3c9e838aa453': '2019-06-16T09:45:00',
      '3ea1c27a-b0e5-4e25-af5e-8e0f9ad159a2': '2019-06-16T10:00:00',
      '426c2477-5cda-44a1-9354-14c33c5e90a7': '2019-06-16T10:15:00'
    };
    const wrapper = shallow(<TimeSlots availableSlots={schedules} />);
    expect(wrapper).toMatchSnapshot();
  });
});
