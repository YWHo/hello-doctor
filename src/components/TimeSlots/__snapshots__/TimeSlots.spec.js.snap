// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSlots render "No Appointment Available" 1`] = `
<styled.div>
  <styled.div>
    No appointment available for this date
  </styled.div>
</styled.div>
`;

exports[`TimeSlots render based on the given schedule 1`] = `
<styled.div>
  <styled.div>
    <styled.button
      key="tb_0"
      onClick={[Function]}
      selected={false}
    >
      <styled.div>
        9:45
      </styled.div>
    </styled.button>
    <styled.button
      key="tb_1"
      onClick={[Function]}
      selected={false}
    >
      <styled.div>
        10:00
      </styled.div>
    </styled.button>
    <styled.button
      key="tb_2"
      onClick={[Function]}
      selected={false}
    >
      <styled.div>
        10:15
      </styled.div>
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`TimeSlots render time selection based on the given schedule 1`] = `
<styled.div>
  <styled.div>
    <styled.button
      key="tb_0"
      onClick={[Function]}
      selected={false}
    >
      <styled.div>
        9:45
      </styled.div>
    </styled.button>
    <styled.button
      key="tb_1"
      onClick={[Function]}
      selected={true}
    >
      <styled.div>
        10:00
      </styled.div>
    </styled.button>
    <styled.button
      key="tb_2"
      onClick={[Function]}
      selected={false}
    >
      <styled.div>
        10:15
      </styled.div>
    </styled.button>
  </styled.div>
</styled.div>
`;
